pkg/stack/types.go

```diff
@@ -25,9 +27,30 @@ const (
 
 var Providers = []string{Aws, Azure, Gcp, Digitalocean}
 
+type FunctionConfig struct {
+	Memory  *int `yaml:"memory,omitempty"`
+	Timeout *int `yaml:"timeout,omitempty"`
+}
+
 type Config struct {
-	Name     string                 `yaml:"name,omitempty"`
-	Provider string                 `yaml:"provider,omitempty"`
-	Region   string                 `yaml:"region,omitempty"`
-	Extra    map[string]interface{} `yaml:",inline,omitempty"`
+	Name     string                    `yaml:"name,omitempty"`
+	Provider string                    `yaml:"provider,omitempty"`
+	Region   string                    `yaml:"region,omitempty"`
+	Config   map[string]FunctionConfig `yaml:"config,omitempty"`
+	Extra    map[string]interface{}    `yaml:",inline,omitempty"`
+}
+
+func DefaultFunctionConfig(provider string) *FunctionConfig {
+	ret := &FunctionConfig{
+		Timeout: to.IntPtr(15),
+	}
+
+	switch provider {
+	case Gcp:
+		ret.Memory = to.IntPtr(512)
+	default:
+		ret.Memory = to.IntPtr(128)
+	}
+
+	return ret
```

Comment:
Would it be possible to shift this config into the providers themselves so they can set their own defaults in the case where its not defined in the configuration?